modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/commands/openCategorySelectionModal.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {addParams, navigate, openSelectionModal} from '../../../index.es';
```

Comment:
Please import with direct path to files, instead of `index`. Doing it this way adds an unnecessary layer of wiring, and we are consistent with direct paths with other imports. I am not sure, but doing import this way might have odd interactions with TS, and a small performance hit.

Speaking of TS, you can use it on the files, and add a `@ts-ignore` on non-TS imports, like [we're doing here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/hooks/useId.ts#L8). I believe you do need to add the exports to both index.d.ts (for importing from TS files) and index.es.js (for importing from JS files). It's technically not a blocker for you to add it in JS, but we will eventually need to migrate it all to TS, so doing it that way if much appreciated :slightly_smiling_face:

Reply:
Done! thanks for the recommendation 😄 . I have added the exports in both files, index.d.ts and index.es.js.